hw/ip/spi_host/data/spi_host.hjson

```diff
@@ -0,0 +1,590 @@
+          resval: "0x0"
+        },
+        { bits:  "23:16",
+          name:  "BYTE_C",
+          desc:  '''One of four bytes to transmit.  If ByteOrder = 0,
+                    this byte will be transmitted second, after BYTE_D.
+                    Otherwise BYTE_C will be transmitted third, after
+                    BYTE_A, and BYTE_B.
+                    ''',
+          resval: "0x0"
+        },
+        { bits:  "15:8",
+          name:  "BYTE_B",
+          desc:  '''One of four bytes to transmit.  If ByteOrder = 0,
+                    this byte will be transmitted third after BYTE_D and BYTE_C.
+                    Otherwise BYTE_B will be transmitted second, after BYTE_A.
+                    ''',
+          resval: "0x0"
+        },
+        { bits:  "7:0",
+          name:  "BYTE_A",
+          desc:  '''One of four bytes to transmit.  If ByteOrder = 0,
+                    This byte will be transmitted last, after BYTE_D, BYTE_C
+                    and BYTE_B.  Otherwise BYTE_A will be transmitted first.
+                    ''',
+          resval: "0x0"
+        }
+      ]
+    },
+    { name: "RXDATA",
+      desc: "SPI Receive data",
+      swaccess: "ro",
+      hwaccess: "hwo",
+      fields: [
+        { bits:  "31:24",
+          name:  "BYTE_D",
+          desc:  '''One of four bytes to receive.  If ByteOrder = 0,
+                    this byte will be received before BYTE_C, BYTE_B
+                    and BYTE_A. Otherwise BYTE_D will be received
+                    last of the four.
+                    ''',
+          resval: "0x0"
+        },
+        { bits:  "23:16",
+          name:  "BYTE_C",
+          desc:  '''One of four bytes to receive.  If ByteOrder = 0,
+                    this byte will be received second, after BYTE_D.
+                    Otherwise BYTE_C will be received third, after
+                    BYTE_A, and BYTE_B.
+                    ''',
+          resval: "0x0"
+        },
+        { bits:  "15:8",
+          name:  "BYTE_B",
+          desc:  '''One of four bytes to receive.  If ByteOrder = 0,
+                    this byte will be received third after BYTE_D
+                    and BYTE_C.  Otherwise BYTE_B will be received
+                    second, after BYTE_A.
+                    ''',
+          resval: "0x0"
+        },
+        { bits:  "7:0",
+          name:  "BYTE_A",
+          desc:  '''One of four bytes to receive.  If ByteOrder = 0,
+                    This byte will be received last, after BYTE_D, BYTE_C
+                    and BYTE_B.  Otherwise BYTE_A will be received first.
+                    ''',
+          resval: "0x0"
+        }
+      ]
+    },
+    { name: "ERROR_ENABLE",
+      desc: "Controls which classes of errors raise an interrupt."
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        { bits: "0",
+          name: "CMDERR",
+          desc: '''Command Error: Assert to send an error interrupt whenever
```

Comment:
Because "assert" can also be a noun, maybe this would be better as "Set this to send ..." ? But, since it's 1 on reset, maybe this should be something like "If this bit is set, the block sends an error interrupt whenever ..." ?

Also, it would be nice to add links to the other registers with the !! syntax.

Reply:
Changed.  

I'll come back to the !!'s when I go back to reviewing the processed output.